src/hotspot/share/runtime/handshake.cpp

```diff
@@ -488,10 +488,11 @@ void HandshakeState::remove_op(HandshakeOperation* op) {
 bool HandshakeState::process_by_self(bool allow_suspend) {
   assert(Thread::current() == _handshakee, "should call from _handshakee");
   assert(!_handshakee->is_terminated(), "should not be a terminated thread");
-  assert(_handshakee->thread_state() != _thread_blocked, "should not be in a blocked state");
-  assert(_handshakee->thread_state() != _thread_in_native, "should not be in native");
 
-  ThreadInVMForHandshake tivm(_handshakee);
+  _handshakee->frame_anchor()->make_walkable(_handshakee);
```

Comment:
I removed it because has_last_Java_frame() just checks if the _last_Java_sp != NULL, but make_walkable() already checks for that at the beginning (in the platforms where we actually do something).